x/ibc/core/legacy/v100/genesis.go

```diff
@@ -0,0 +1,54 @@
+package v100
+
+import (
+	"github.com/line/lbm-sdk/client"
+	"github.com/line/lbm-sdk/codec"
+	genutiltypes "github.com/line/lbm-sdk/x/genutil/types"
+	tmtypes "github.com/line/ostracon/types"
+
+	clientv100 "github.com/line/lbm-sdk/x/ibc/core/02-client/legacy/v100"
+	clienttypes "github.com/line/lbm-sdk/x/ibc/core/02-client/types"
+	connectiontypes "github.com/line/lbm-sdk/x/ibc/core/03-connection/types"
+	host "github.com/line/lbm-sdk/x/ibc/core/24-host"
+	"github.com/line/lbm-sdk/x/ibc/core/types"
+)
+
+// MigrateGenesis accepts exported v1.0.0 IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v1 to v2)
+// - Remove all solo machine consensus states
+// - Remove all expired tendermint consensus states
+func MigrateGenesis(appState genutiltypes.AppMap, clientCtx client.Context, genDoc tmtypes.GenesisDoc, maxExpectedTimePerBlock uint64) (genutiltypes.AppMap, error) {
+	if appState[host.ModuleName] != nil {
+		// ensure legacy solo machines are registered
+		clientv100.RegisterInterfaces(clientCtx.InterfaceRegistry)
+
+		// unmarshal relative source genesis application state
+		ibcGenState := &types.GenesisState{}
+		clientCtx.Codec.MustUnmarshalJSON(appState[host.ModuleName], ibcGenState)
```

Comment:
Because lint say clientCtx.JSONCodec is deprecated, it use clientCtx.Codec. 
https://github.com/line/lbm-sdk/runs/8200263100?check_suite_focus=true#step:5:44

The latest ibc-go also use clientCtx.Codec, that it is not stable version.
https://github.com/cosmos/ibc-go/blob/v5.0.0-rc1/modules/core/legacy/v100/genesis.go